### PR TITLE
Feature: Azure Network Watcher

### PR DESCRIPTION
**Pull request by vg-svitla**

### CODE REVIEW

File: Azure/connectors/azure_blob.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        for line in data.get("records", []):
+            line_time = isoparse(line["time"]).astimezone(timezone.utc)
+
+            # If the record is too old, ignore it.
+            if time_filter and line_time < time_filter:
+                continue
+
+            result.append(line)
+
+        return result
```
Comment (squioc) [inline Azure/connectors/azure_blob.py:158]: The content of the blob file can be textual. I suggest removing this part:
```suggestion
```

File: Azure/connectors/azure_blob.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                        )
+
+                    await delete_file(file)
+
+                if content:
+                    records.extend(self.filter_blob_data(orjson.loads(content), self.last_event_date))
+
+        result: list[str] = await self.push_data_to_intakes(
+            [orjson.dumps(event).decode("utf-8") for event in records],
+        )
```
Comment (squioc) [inline Azure/connectors/azure_blob.py:124]: according to the previous comment:
```suggestion
        result: list[str] = await self.push_data_to_intakes(events=records)
```

File: Azure/trigger_azure_blob_storage_events.json
Comment (squioc) [inline Azure/trigger_azure_blob_storage_events.json:1]: Can you transfer the uuid of this manifest to the `trigger_azure_network_watcher_collector.json` and change the one here? or reverse the identifier of the two manifests in order that the existing connector are associated to Azure Network Watcher? Do the name in the `connector_*` manifest. Thanks

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for the PR.
Some changes to apply

File: Azure/connectors/azure_blob.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                            file_content = decompress(file_content)
+
+                        records.extend(
+                            self.filter_blob_data(orjson.loads(file_content.decode("utf-8")), self.last_event_date)
+                        )
+
+                    await delete_file(file)
+
+                if content:
+                    records.extend(self.filter_blob_data(orjson.loads(content), self.last_event_date))
```
[the remark this answers is not in the document]
Comment (vg-svitla) [reply Azure/connectors/azure_blob.py]: I just think we can split by `\n` the content. Otherwise we will push entire file content as 1 event

**Comment by rombernier** [comment]
Hello, can you merge this fix soon?, it's requested by the client https://github.com/SekoiaLab/platform/issues/53514 Thanks

File: Azure/connector_azure_network_watcher_collector.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      "intake_key"
+    ],
+    "secrets": [
+      "account_key"
+    ]
+  },
+  "description": "connector playbook to get Azure Network Watcher events information",
+  "docker_parameters": "azure_network_watcher",
+  "name": "[BETA] Collect Azure Network Watcher events",
+  "uuid": "3e5ff414-f54a-4c3e-8127-82177b31e3c4"
```
Comment (Adamowoc) [inline Azure/connector_azure_network_watcher_collector.json:47]:
```suggestion
  "uuid": "626f4d84-5ab7-11ee-8c99-0242ac120002"
```
Azure network watcher should take the old uuid of Azure blob storage

File: Azure/trigger_azure_network_watcher_collector.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    },
+    "description": "Trigger playbook to get Azure Network Watcher events information",
+    "docker_parameters": "azure_network_watcher",
+    "name": "[BETA] Fetch new Azure Network Watcher events from Blob Storage",
+    "results": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "properties": {},
+        "title": "The records"
+    },
+    "uuid": "3e5ff414-f54a-4c3e-8127-82177b31e3c4"
```
Comment (Adamowoc) [inline Azure/trigger_azure_network_watcher_collector.json:52]:
```suggestion
    "uuid": "626f4d84-5ab7-11ee-8c99-0242ac120002"
```
Same here

File: Azure/connector_azure_blob_storage_events.json [hunk 6]
```diff
@@ -41,7 +41,7 @@
       "account_key"
     ]
   },
-  "description": "connector p laybook to get Azure Blob Storage events information",
+  "description": "connector playbook to get Azure Blob Storage events information",
   "docker_parameters": "azure_blob_storage",
   "name": "[BETA] Collect Azure Blob Storage events",
   "uuid": "626f4d84-5ab7-11ee-8c99-0242ac120002"
```
Comment (Adamowoc) [inline Azure/connector_azure_blob_storage_events.json:47]: Azure blob storage needs a new uuid

File: Azure/trigger_azure_blob_storage_events.json [hunk 7]
```diff
@@ -41,7 +41,7 @@
             "account_key"
         ]
     },
-    "description": "Trigger p laybook to get Azure Blob Storage events information",
+    "description": "Trigger playbook to get Azure Blob Storage events information",
     "docker_parameters": "azure_blob_storage",
     "name": "[BETA] Collect Azure Blob Storage events",
     "results": {
```
Comment (Adamowoc) [inline Azure/trigger_azure_blob_storage_events.json:47]: modify the new uuid of Azure blob storage in this file too

**Review by Adamowoc — CHANGES_REQUESTED** [review]
Do not forget to update the CHANGELOG.md

**Review by Adamowoc — CHANGES_REQUESTED** [review]

**Review by squioc — APPROVED** [review]

**Review by Adamowoc — APPROVED** [review]

**Pull request merged**